File: occasum-core/src/main/java/pw/stamina/occasum/properties/PropertyParseException.java
package pw.stamina.occasum.properties;

public final class PropertyParseException extends Exception {
    private final String input;

    public PropertyParseException(String message, String input) {
        super(message);
        this.input = input;
    }

    public PropertyParseException(String message, Throwable cause, String input) {
        super(message, cause);
        this.input = input;
    }

    public PropertyParseException(Throwable cause, String input) {
        super(cause);
        this.input = input;
    }

    public static PropertyParseException nullInput() {
        return new PropertyParseException("input is null", null);
    }

    public String getInput() {
        return input;
    }
}

File: occasum-core/src/main/kotlin/pw/stamina/occasum/properties/PropertyParseException.kt
package pw.stamina.occasum.properties

class PropertyParseException : Exception {
    val input: String

    constructor(message: String, input: String) : super(message) {
        this.input = input
    }

    constructor(message: String, cause: Throwable, input: String) : super(message, cause) {
        this.input = input
    }

    constructor(cause: Throwable, input: String) : super(cause) {
        this.input = input
    }
}
